core/dbt/contracts/graph/manifest.py

```diff
@@ -1011,25 +1011,52 @@ def resolve_doc(
                 return result
         return None
 
+    def merge_from_artifact(
+        self,
+        other: 'WritableManifest',
+        selected: Set[UniqueID],
+    ) -> None:
+        """Given the selected unique IDs and a writable manifest, update this
+        manifest by replacing any unselected nodes with their counterpart.
+
+        Only non-ephemeral refable nodes are examined.
+        """
+        refables = set(NodeType.refable())
+        merged = set()
+        for unique_id, node in other.nodes.items():
+            if (
+                node.resource_type in refables and
+                not node.is_ephemeral and
+                unique_id not in selected
+            ):
+                merged.add(unique_id)
+                self.nodes[unique_id] = node
+
```

Comment:
Where is the logic to flip on the `deferred` flag? I see that `deferred: True` in several of the test cases. When I ran with deferral locally and check the manifest afterward, `deferred` is always false.